### PR TITLE
Allow Microsoft.WinGet.Client to run in any PowerShell session running as system

### PR DESCRIPTION
**Pull request by msftrubengu**
This PR adds support for running the Microsoft.WinGet.Client module in system context without the need to start pwsh.exe -MTA. Running as MTA is required using inproc Microsoft.Management.Deployment.

If the module is running inproc and the current thread is not an MTA, it will create a new MTA thread and execute there. Otherwise, non inproc or already an MTA will use the current thread.

This was done by sharing AsyncCommand (renamed to PowerShellCmdlet) from Microsoft.WinGet.Configuration. Originally, I wanted to create a new shared lib, but decided to just share the files between the projects.

All cmdlets must inherit from PowerShellCmdlet. All cmdlets that use Microsoft.Management.Deployment must inherit ManagementDeploymentCommand and use Execute at the command engine entry point. As a safe mechanism, if any call to PackageManagerWrapper will verify the thread is not an STA if running inproc and fail.

I verified the cmdlets work in system context locally. A future PR will add running our pester tests using psexe.exe

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/3816)

### CODE REVIEW

File: src/PowerShell/Microsoft.WinGet.Client.Engine/Commands/Common/FinderCommand.cs [hunk 1]
```diff
@@ -35,31 +35,33 @@ internal FinderCommand(PSCmdlet psCmdlet)
...
         /// </summary>
-        protected string Source { get; set; }
+        protected string? Source { get; set; }
 
         /// <summary>
         /// Gets or sets how to match against package fields.
         /// </summary>
-        protected string[] Query { get; set; }
+#pragma warning disable SA1011 // Closing square brackets should be spaced correctly
+        protected string[]? Query { get; set; }
```
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.Client.Engine/Commands/Common/FinderCommand.cs:61]: Academic question: Does `string?[]` mean a non-null array of nullable strings, while `string[]?` means a nullable array of non-null strings?  If so, then `string?[]?` is a lovingly confusing type name 😄

File: src/PowerShell/Microsoft.WinGet.Client.Engine/Helpers/PackageManagerWrapper.cs [hunk 2]
```diff
@@ -21,7 +22,7 @@ internal sealed class PackageManagerWrapper
     {
         private static readonly Lazy<PackageManagerWrapper> Lazy = new (() => new PackageManagerWrapper());
 
-        private PackageManager packageManager = null;
+        private PackageManager? packageManager = null;
```
Comment (JohnMcPMS) [inline src/PowerShell/Microsoft.WinGet.Client.Engine/Helpers/PackageManagerWrapper.cs:25]: What is the point of making this nullable and then just overriding that everywhere?  Can you not do `null!` to trick it into letting you have a null value?  Maybe you need to do `((PackageManager?)null)!`.

**Review by JohnMcPMS — DISMISSED** [review]

File: src/PowerShell/Microsoft.WinGet.Client.Engine/Helpers/PackageManagerWrapper.cs (hunk 2 above)
Comment (msftrubengu) [reply src/PowerShell/Microsoft.WinGet.Client.Engine/Helpers/PackageManagerWrapper.cs]: `null!` is the way to go

File: src/PowerShell/Microsoft.WinGet.Client.Engine/Commands/Common/FinderCommand.cs (hunk 1 above)
Comment (msftrubengu) [reply src/PowerShell/Microsoft.WinGet.Client.Engine/Commands/Common/FinderCommand.cs:61]: Yup. There's another place where I needed a non-nullable array with nullable objects, so it was object?[]

**Review by JohnMcPMS — APPROVED** [review]

**Pull request merged**